Source/Scene/Expression.js

```diff
@@ -0,0 +1,1904 @@
+                return property.z;
+            } else if (member === 'a') {
+                return property.w;
+            }
+        }
+        return property[member];
+    };
+
+    Node.prototype._evaluateMemberBrackets = function(frameState, feature) {
+        if (checkFeature(this._left)) {
+            return feature.getProperty(this._right.evaluate(frameState, feature));
+        }
+        var property = this._left.evaluate(frameState, feature);
+        if (!defined(property)) {
+            return undefined;
+        }
+
+        var member = this._right.evaluate(frameState, feature);
+        if ((property instanceof Cartesian2) || (property instanceof Cartesian3) || (property instanceof Cartesian4)) {
+            // Vector components may be accessed with [0][1][2][3], ['r']['g']['b']['a'] and implicitly with ['x']['y']['z']['w']
+            // For Cartesian2 and Cartesian3 out-of-range components will just return undefined
+            if (member === 0 || member === 'r') {
+                return property.x;
+            } else if (member === 1 || member === 'g') {
+                return property.y;
+            } else if (member === 2 || member === 'b') {
+                return property.z;
+            } else if (member === 3 || member === 'a') {
+                return property.w;
+            }
+        }
+        return property[member];
+    };
+
+    Node.prototype._evaluateArray = function(frameState, feature) {
+        var array = [];
+        for (var i = 0; i < this._value.length; i++) {
+            array[i] = this._value[i].evaluate(frameState, feature);
+        }
+        return array;
+    };
+
+    // PERFORMANCE_IDEA: Have "fast path" functions that deal only with specific types
+    // that we can assign if we know the types before runtime
+
+    Node.prototype._evaluateNot = function(frameState, feature) {
+        var left = this._left.evaluate(frameState, feature);
+        //>>includeStart('debug', pragmas.debug);
+        if (typeof left !== 'boolean') {
+            throw new DeveloperError('Operator "!" requires a boolean argument. Argument is ' + left + '.');
+        }
+        //>>includeEnd('debug');
+        return !left;
+    };
+
+    Node.prototype._evaluateNegative = function(frameState, feature) {
+        var left = this._left.evaluate(frameState, feature);
+        if (left instanceof Cartesian2) {
+            return Cartesian2.negate(left, ScratchStorage.getCartesian2());
+        } else if (left instanceof Cartesian3) {
+            return Cartesian3.negate(left, ScratchStorage.getCartesian3());
+        } else if (left instanceof Cartesian4) {
+            return Cartesian4.negate(left, ScratchStorage.getCartesian4());
+        } else if (typeof left === 'number') {
+            return -left;
+        }
+
+        //>>includeStart('debug', pragmas.debug);
+        throw new DeveloperError('Operator "-" requires a vector or number argument. Argument is ' + left + '.');
```

Comment:
As discussed in #5372, all of these should probably be `RuntimeError` and not be pragma'd out, which means the return statements below them can also be removed.